### DISCUSSION TITLE
How to get list of all user with each roles

### DISCUSSION THREAD

**Discussion by ridlafalah** [spatie/laravel-permission#1678] [Q&A] [answered]
hai, im newbie for spatie laravel permission, as i mention on the title, i dont know how to get list of user with each roles

Name | Username | Role
Emily | Emily22 | Teacher
John | John33 | Student

as far as i tried, only get the name and username, i dont know how to get roles and display it in 1 table

thanks,

**Comment by drbyte** [comment, marked as answer]
Unlike a username, users can have more than one role, so you'll need to deal with the collection.

```
$users = User::with(['roles'])->get();

foreach($users as $user) {
  // $user->name;
  // $user->username;
  // $user->getRoleNames(); // collection of names of roles assigned to the user
     // $roles = implode(', ', $user->getRoleNames()); // comma-separated list of role names
}
```

These are all Eloquent features, so normal Laravel Eloquent capabilities exist.
See https://spatie.be/docs/laravel-permission/v4/basic-usage/basic-usage#eloquent
